determined-ai/determined
# chore: Remove obsolete workloads from Trials API

## Description

The Trial API has been slow on large experiments because it includes every metric on every workload.  With the updated Workloads API ( #4407 ), and downsampling for our chart, we removed our reliance on `trial.workloads` for metrics.

**This PR removes the full workloads response from the Trial API**.  It finishes the last 3 remaining issues: metric names, workloadCount (to know if an experiment successfully recorded 1 metric) and totalCheckpointSize (already included in Trial API).

## Test Plan

- Run a single-trial experiment. On its experiment page, "Total Checkpoint Size" field should be populated (1.22 MB for Titanic const). You should see all Training and Validation metrics selectable for the chart.
- Run a multi-trial experiment. Click the "Trials" tab. In the table, use the checkboxes to select 2+ trials. Select an action... Compare Trials.  You should see checkpoint size > 0, and all training and validation metrics listed.

## Checklist

See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
